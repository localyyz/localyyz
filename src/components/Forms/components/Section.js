import React from "react";
import { View, StyleSheet, Text } from "react-native";

// third party
import PropTypes from "prop-types";

// custom
import { Styles, Sizes, Colours } from "localyyz/constants";

export default class BaseField extends React.Component {
  static propTypes = {
    style: PropTypes.object,
    hasMargin: PropTypes.bool,
    label: PropTypes.string
  };

  static defaultProps = {
    hasMargin: true
  };

  render() {
    return (
      <View
        style={[
          styles.container,
          this.props.style,
          this.props.hasMargin && styles.containerMargin
        ]}>
        <View style={styles.header}>
          <Text style={styles.label}>{this.props.label}</Text>
        </View>
        <View style={styles.content}>{this.props.children}</View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: Sizes.InnerFrame,
    backgroundColor: Colours.Foreground
  },

  containerMargin: {
    marginBottom: Sizes.InnerFrame / 2
  },

  header: {
    paddingHorizontal: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 2
  },

  label: {
    ...Styles.Text,
    ...Styles.SmallText,
    ...Styles.Emphasized
  },

  content: {}
});
