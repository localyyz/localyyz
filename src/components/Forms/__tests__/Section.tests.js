import React from "react";
import Renderer from "react-test-renderer";

// local
import Section from "../components/Section";

// constants
const NAME = "Section Test - Form";
const PROPS = {
  label: "Sample Label",
  style: {}
};

const Component = Section;

describe(NAME, () => {
  it(`${NAME}: should render properly`, () => {
    let c = Renderer.create(<Component {...PROPS} />).getInstance();
    expect(c).toBeDefined();
  });

  it(`${NAME}: should render to snapshot: hasMargin=true`, () => {
    PROPS.hasMargin = true;
    expect(
      Renderer.create(<Component {...PROPS} />).toJSON()
    ).toMatchSnapshot();
  });

  it(`${NAME}: should render to snapshot: hasMargin=false`, () => {
    PROPS.hasMargin = false;
    expect(
      Renderer.create(<Component {...PROPS} />).toJSON()
    ).toMatchSnapshot();
  });
});
