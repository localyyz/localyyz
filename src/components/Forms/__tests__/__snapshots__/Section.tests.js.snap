// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Test - Form Section Test - Form: should render to snapshot: hasMargin=false 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "paddingVertical": 7.5,
      },
      Object {},
      false,
    ]
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 25,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
      Sample Label
    </Text>
  </View>
  <View
    style={Object {}}
  />
</View>
`;

exports[`Section Test - Form Section Test - Form: should render to snapshot: hasMargin=true 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "paddingVertical": 7.5,
      },
      Object {},
      Object {
        "marginBottom": 7.5,
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 25,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
      Sample Label
    </Text>
  </View>
  <View
    style={Object {}}
  />
</View>
`;
